File: frontend/src/api/feedback.js
// 건의 창구 API.
import axios from "axios";

import client from "./client";

const BASE = import.meta.env.VITE_API_URL || "http://localhost:8000";

// anonymous=true면 토큰 없이 보내 백엔드가 작성자 정보 없이 저장(get_optional_user→None).
// false면 로그인 토큰이 실려 실명으로 전달된다.
export const sendFeedback = (content, anonymous = false) =>
  anonymous
    ? axios.post(`${BASE}/api/feedback`, { content }).then((r) => r.data)
    : client.post("/api/feedback", { content }).then((r) => r.data);

// ─── 관리자 전용 ───
export const getFeedback = () => client.get("/api/feedback").then((r) => r.data);

export const markFeedbackRead = (id) =>
  client.patch(`/api/feedback/${id}/read`).then((r) => r.data);

export const deleteFeedback = (id) =>
  client.delete(`/api/feedback/${id}`).then((r) => r.data);

File: backend/app/routers/feedback.py
"""feedback 라우터 — 방문자 건의 접수(익명 허용) + 관리자 조회/읽음/삭제."""
from fastapi import APIRouter, Depends, HTTPException

from app.auth import get_optional_user, require_admin
from app.database import supabase
from app.models.schemas import FeedbackCreate

router = APIRouter()


@router.post("", status_code=201)
def create_feedback(body: FeedbackCreate, user: dict | None = Depends(get_optional_user)):
    """건의 접수. 로그인 상태면 이름/유저 연결, 아니면 익명."""
    content = body.content.strip()
    if not content:
        raise HTTPException(status_code=400, detail="내용을 입력해주세요.")
    supabase.table("feedback").insert(
        {
            "content": content,
            "author_name": user["name"] if user else None,
            "user_id": user["id"] if user else None,
        }
    ).execute()
    return {"ok": True}


@router.get("")
def list_feedback(_: dict = Depends(require_admin)):
    """전체 건의 (최신순) — 관리자 대시보드용."""
    return (
        supabase.table("feedback")
        .select("*")
        .order("created_at", desc=True)
        .execute()
        .data
    )


@router.patch("/{feedback_id}/read")
def mark_read(feedback_id: str, _: dict = Depends(require_admin)):
    supabase.table("feedback").update({"is_read": True}).eq("id", feedback_id).execute()
    return {"ok": True}


@router.delete("/{feedback_id}")
def delete_feedback(feedback_id: str, _: dict = Depends(require_admin)):
    supabase.table("feedback").delete().eq("id", feedback_id).execute()
    return {"ok": True}
